### ISSUE TITLE
CoreWebView2CreateCoreWebView2ControllerCompletedHandler called with NULL controller on Windows 11

### ISSUE THREAD

**Issue by rlzack** [MicrosoftEdge/WebView2Feedback#2687]
I have an embedded WebView2 in my application (based mostly on various examples here and on docs.microsoft.com) which works quite well. But I have found that it crashes my application on Windows 11. Some simple debugging has shown that when the CoreWebView2CreateCoreWebView2ControllerCompletedHandler is called, it is called with a first argument (HRESULT hr) of E_UNEXPECTED and a second argument (ICoreWebView2Controller* controller) of NULL.

How can I figure out what is going wrong here?

My application is written in C++ using VS 2022, and I'm willing to provide any more information about my environment. I'm not sure what else might be needed.

**Comment by rlzack** [comment]
Oops - this may be my error. Is WebView2 preinstalled on Windows 11? It was on my dev image, and am asking others in my organization.

**Comment by victorthoang** [comment]
@rlzack yes, webview2 is included in windows 11. 

There is a [blog post](https://blogs.windows.com/msedgedev/2022/06/27/delivering-the-microsoft-edge-webview2-runtime-to-windows-10-consumers/#:~:text=Starting%20with%20Windows%2011%2C%20the,the%20runtime%20with%20their%20applications.) if you're interested. It is currently being rolled out to windows 10 devices via a push update with much success.

**Comment by rlzack** [comment]
This is wonderful. But now I need to know how to detect that there is a WebView2 runtime installed on the machine, so know not to try to install one myself. I feel like this is a stupid question - and should have a well known answer. But I have searched without luck.

**Comment by LiangTheDev** [comment]
You can call [GetAvailableCoreWebView2BrowserVersionString](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/webview2-idl#getavailablecorewebview2browserversionstring) and evaluate whether the versionInfo is nullptr. More info at https://docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/distribution#detect-if-a-suitable-webview2-runtime-is-already-installed.

**Comment by rlzack** [comment]
Thanks for the info, @LiangTheDev . I think this will work for us.

Sadly, I am not going to be able to provide a performance trace for you. But note that this happened with an older WebView2 runtime that we installed with our app, and which we used when creating the WebView2 environment. The way I got around the E_UNEXPECTED was to use the built-in version of WebView2 runtime, rather than the version in our distribution. As an older runtime, I doubt that it will be much use to you. And it would take quite a bit of time for me to set up that environment again.
